runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/FnApiControlClientPoolService.java

```diff
@@ -99,9 +99,6 @@ public static FnApiControlClientPoolService offeringClientsToPool(
       // We do not attempt to transactionally add the client to our internal list and offer it to
       // the sink.
       clientSink.put(headerAccessor.getSdkWorkerId(), newClient);
-    } catch (InterruptedException e) {
-      Thread.currentThread().interrupt();
-      throw new RuntimeException(e);
```

Comment:
Strictly speaking, removing this changes the logic; if any higher-level code catches the exception, the interrupted flag will not be set.

Reply:
This exception catch was added in [2017](https://github.com/apache/beam/commit/9ed655be780630e1218d185bd0d2ebfea099b988#diff-40daa607c5288b4785dada738d9de8ecR31). The purpose was to catch InterruptedException from `BlockingQueue # put (E var1) throws InterruptedException`. At present, the implementation of the code has changed from BlockingQueue to [ControlClientPool.Sink # put](https://github.com/apache/beam/blob/59b4b6d5318594c6fd7fae2dd6be2dd8aca972fa/runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/FnApiControlClientPoolService.java#L40). There is no chance of throwing an `InterruptedException`. So I think it is possible to delete the logic of this exception catch. Am i missing something here？

Reply:
I see. This kind of issue is easyly to neglect. :)